STANDALONE_R.md

```diff
@@ -0,0 +1,27 @@
+
+Other paths can be added to the `.libPaths()` through other means. An overview of different config options can be found in a [Posit help article about R config options](https://support.posit.co/hc/en-us/articles/360047157094-Managing-R-with-Rprofile-Renviron-Rprofile-site-Renviron-site-rsession-conf-and-repos-conf). For us, the most relevant file to alter user behavior is `~/.Rprofile`, which can contain R code that will be sourced to modify R's behavior. Here, we can modify the `.libPaths()` with a line of R code, like `.libPaths(c("/path/to/existing/folder", .libPaths()))`. The R session will use that path to look for libraries, allowing support staff and instructors to install packages to the course shared folder, and students to access the path with a single line of terminal code: `echo '.libPaths(c("/path/to/existing/folder", .libPaths()))' > ~/.Rprofile`
+
+In the case of BST 262, the command to create an .Rprofile file is
+
+```bash
+echo '.libPaths(c("/shared/courseSharedFolders/144013outer/144013/R/x86_64-pc-linux-gnu-library/4.4", .libPaths()))' > ~/.Rprofile
+```
+
+## R Package Installs
+
+R package installs can work in a user home directory, or in a shared directory when the `libPaths()` are set up properly. However, R package installs can also have external dependencies. These external dependencies will come up when installing a package, like the need for `gdal` when installing the R package `raster`. In that case, when R is installed with Spack, the additional depencencies should also be installed with Spack to the same environment where R is installed, providing access to the software.
+
+However, sometimes this isn't enough. In the case of installing `raster`, the dependent package `terra` failed to install because it wasn't able to locate the library files for either `libproj` or `sqlite3`, both of which were installed as Spack packages to the environment via the spack packages `proj` and `sqlite` (see the bst262 spack environment file). I found a [GitHub issue with other people describing the same problem](https://github.com/r-spatial/sf/issues/1471#issuecomment-2148767638), and adapted the install line to `install.packages('terra', configure.args=c("--with-sqlite3-lib=/shared/spack/opt/spack/linux-amzn2-skylake_avx512/gcc-14.1.0/sqlite-3.46.0-qxr7hpygeupogc6rx745sfsfa2ykhws4/lib"))`. It seems that when using `install.packages()` in R, the `configure.args` argument can be used to pass flags to the installer. I got the flag from the GitHub issue, so I'm not sure if there's a way to determine what flags are available in other installers. Another option might be modifying the `LD_LIBRARY_PATHS` variable outside of R, if a similar issue arises and a similar flag for the installer can't be found. In any case, installing the problem dependency with the additional flag worked, which allowed the `raster` package to be installed. Note that passing the additional flag directly to the `raster` install didn't work.
```

Comment:
Thanks for calling out this gotcha with dependent packages installed in the Spack environment and using a pattern like `install.packages('package', configure.args=c("--with-my-lib=/path/to/lib"))` to handle it.

For future reference, you can identify what options are available for a package like `terra` by doing downloading the source and invoking the `configure` script which is used prior to running `make`.  For example:


1. Download the `terra` repo:
```
$ git clone https://github.com/rspatial/terra.git
Cloning into 'terra'...
remote: Enumerating objects: 103588, done.
remote: Counting objects: 100% (2250/2250), done.
remote: Compressing objects: 100% (53/53), done.
remote: Total 103588 (delta 2222), reused 2197 (delta 2197), pack-reused 101338 (from 1)
Receiving objects: 100% (103588/103588), 190.69 MiB | 38.41 MiB/s, done.
Resolving deltas: 100% (93210/93210), done.
Updating files: 100% (528/528), done.
```
2. Invoke `./configure --help` to see all options, including `--with-sqlite3-lib=LIB_PATH`:
```
$ cd terra
$ ./configure --help
'configure' configures this package to adapt to many kinds of systems.

Usage: ./configure [OPTION]... [VAR=VALUE]...

To assign environment variables (e.g., CC, CFLAGS...), specify them as
VAR=VALUE.  See below for descriptions of some of the useful variables.

Defaults for the options are specified in brackets.

Configuration:
  -h, --help              display this help and exit
      --help=short        display options specific to this package
      --help=recursive    display the short help of all the included packages
  -V, --version           display version information and exit
  -q, --quiet, --silent   do not print 'checking ...' messages
      --cache-file=FILE   cache test results in FILE [disabled]
  -C, --config-cache      alias for '--cache-file=config.cache'
  -n, --no-create         do not create output files
      --srcdir=DIR        find the sources in DIR [configure dir or '..']

Installation directories:
  --prefix=PREFIX         install architecture-independent files in PREFIX
                          [/usr/local]
  --exec-prefix=EPREFIX   install architecture-dependent files in EPREFIX
                          [PREFIX]

By default, 'make install' will install all the files in
'/usr/local/bin', '/usr/local/lib' etc.  You can specify
an installation prefix other than '/usr/local' using '--prefix',
for instance '--prefix=$HOME'.

For better control, use the options below.

Fine tuning of the installation directories:
  --bindir=DIR            user executables [EPREFIX/bin]
  --sbindir=DIR           system admin executables [EPREFIX/sbin]
  --libexecdir=DIR        program executables [EPREFIX/libexec]
  --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
  --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
  --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
  --libdir=DIR            object code libraries [EPREFIX/lib]
  --includedir=DIR        C header files [PREFIX/include]
  --oldincludedir=DIR     C header files for non-gcc [/usr/include]
  --datarootdir=DIR       read-only arch.-independent data root [PREFIX/share]
  --datadir=DIR           read-only architecture-independent data [DATAROOTDIR]
  --infodir=DIR           info documentation [DATAROOTDIR/info]
  --localedir=DIR         locale-dependent data [DATAROOTDIR/locale]
  --mandir=DIR            man documentation [DATAROOTDIR/man]
  --docdir=DIR            documentation root [DATAROOTDIR/doc/PACKAGE]
  --htmldir=DIR           html documentation [DOCDIR]
  --dvidir=DIR            dvi documentation [DOCDIR]
  --pdfdir=DIR            pdf documentation [DOCDIR]
  --psdir=DIR             ps documentation [DOCDIR]

Optional Packages:
  --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
  --without-PACKAGE       do not use PACKAGE (same as --with-PACKAGE=no)
  --with-gdal-config=GDAL_CONFIG
                          the location of gdal-config
  --with-data-copy=yes/no local copy of data directories in package, default
                          no
  --with-proj-data=DIR    location of PROJ data directory
  --with-sqlite3-lib=LIB_PATH
                          the location of sqlite3 libraries
  --with-proj-include=DIR location of proj header files
  --with-proj-api=yes/no  use the deprecated proj_api.h even when PROJ 6 is
                          available; default no
  --with-proj-lib=LIB_PATH
                          the location of proj libraries
  --with-proj-share=SHARE_PATH
                          the location of proj metadata files
  --with-geos-config=GEOS_CONFIG
                          the location of geos-config

Some influential environment variables:
  CC          C compiler command
  CFLAGS      C compiler flags
  LDFLAGS     linker flags, e.g. -L<lib dir> if you have libraries in a
              nonstandard directory <lib dir>
  LIBS        libraries to pass to the linker, e.g. -l<library>
  CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir> if
              you have headers in a nonstandard directory <include dir>

Use these variables to override the choices made by 'configure' or to help
it to find libraries and programs with nonstandard names/locations.

Report bugs to the package provider.
```